### DISCUSSION TITLE
Sometimes >15 seconds of delay between pressing a button and turning on the light

### DISCUSSION THREAD

**Discussion by ikarisan** [Koenkk/zigbee2mqtt#19037] [General]
Hi!

Since many month I have a very annoying problem with my Zigbee network.
I am using some TuYa TS004F switches and also Alexa to control my lights.
The path is

TS004F => Zigbee2Mqtt => Mosquitto => NodeRed => Mosquito => Zigbee2Mqtt
or
Alexa => NodeRed => Mosquitto => Zigbee2Mqtt

The NodeRed flow simply reacts on the button/alexa and sends either

msg.payload = {"state":"ON"}
msg.payload = {"state":"OFF"} or 
msg.payload = {"state":"TOGGLE"}

to the MQTT topic of the light. 

Sometime it happens, that I press the "light on" button and nothing happens.
Then I press the button again and nothing happens.
Then I press the button again and nothing happens.

After 10...15 seconds the light turns on - off - on within 1-2 seconds.

The same happens when I say "Alexa turn the light on"... nothing happens.
If I'm lucky, the light comes on after a few seconds. Sometimes I have to repeat the command.

Once the latency has occurred, the further commands work almost without delay.

I use a TI CC2538 + CC2592 coordinator with zStack30x (version 20201010) and Zigbee2Mqtt 1.33.0-dev.
Zigbee2Mqtt, NodeRed and Mosquitto are running as docker containers on the same system.
A few days ago I replaced my "Raspberry Pi4" with a "Lenovo ThinkCentre M910q" as I thought it might be a resource problem. 
But nothing changed. :( 

I am at a loss. What do I do now? :( 

Thorsten

**Comment by lux73** [comment]
Toggle the Light via Z2M WebIf - if there is no Delay the Problem is NOT on Z2M Side

**Reply by ikarisan** [reply]
The problem is that the problem does not always occur. Sometimes it works great for days.
But I tested it via the web interface => no delay.
Then I sent some {"state": "TOGGLE"} to the MQTT /set toppic => no delay. 
But at the moment every light switches instantly. :(

**Reply by lux73** [reply]
when the Problem is present please test/toggle immediately via WebIf - if it's not present then your Problem is in your Chain:

```
TS004F => Zigbee2Mqtt => Mosquitto => NodeRed => Mosquito => Zigbee2Mqtt
or
Alexa => NodeRed => Mosquitto => Zigbee2Mqtt
```

anywhere

**Comment by ikarisan** [comment]
Maybe I have found a possible reason. In my flat I have two ZigBee routers (TRADFRI transformer 10W) that are not constantly powered because they are part of my 3D printers and only used for lightning when I print. Eventually the switches or the lights are using the Tradfri trafos as part of the route and when I switch of the 3D printer the route has to been build up again.
